traitsui/testing/tester/tests/test_ui_tester.py

```diff
@@ -57,6 +58,23 @@ def test_ui_disposed(self):
             pass
         self.assertTrue(ui.destroyed)
 
+    def test_create_ui_reraise_exception(self):
+        tester = UITester()
+        order = Order()
+        view = View(Item("submit_button"))
+
+        with self.assertRaises(RuntimeError), \
+                self.assertLogs("traitsui", level="ERROR"):
+
+            with tester.create_ui(order, dict(view=view)) as ui:
+
+                def raise_error():
+                    raise ZeroDivisionError()
+
+                GUI().invoke_later(raise_error)
```

Comment:
you say in the PR description: "create_ui context manager should capture and reraise exceptions occurred from the GUI event loop while disposing a UI."  
Just to clarify, because `raise_error` raises a `ZeroDivisionError` but we have `with self.assertRaises(RuntimeError)` we know that the raise_error ultimately triggered the `reraise_exceptions` that was added in the file below (as that 'captures all exceptions occurred in the context and then reraises a RuntimeError if there are any exceptions captured.').  

If thats true this all LGTM, I just want to make sure Im understanding correctly.
Also, do we know anything specific about when a call to `.invoke_later()` will be processed (aside from that it will be later)?

Reply:
> Just to clarify, because raise_error raises a ZeroDivisionError but we have with self.assertRaises(RuntimeError) we know that the raise_error ultimately triggered the reraise_exceptions that was added in the file below (as that 'captures all exceptions occurred in the context and then reraises a RuntimeError if there are any exceptions captured.').

Yes that is correct.
Always reraising a new RuntimeError is intentional, as the opposite of reraising whatever is captured is problematic in this situation. See https://github.com/enthought/traitsui/pull/1099

> Also, do we know anything specific about when a call to .invoke_later() will be processed (aside from that it will be later)?

The callable is queued to be processed by the GUI event loop. So given the way I have written this test, it should occur from the first `process_cascade_events` in the `finally` block. 
If I mock the UI.dispose to call `invoke_later(raise_error)`, then the error will occur from the second `process_cascade_events` in the `finally`. Mocking `UI.dispose` is a bit harder because `UI` is an instance of `HasStrictTraits` and it is not as trivial to mock its method compared to mocking any methods of a plain simple Python object.